# DP: Backport r240457 from trunk

internal/syscall/unix: add getrandom syscall for MIPS and SPARC

Reviewed-on: https://go-review.googlesource.com/29678

Index: b/libgo/go/internal/syscall/unix/getrandom_linux_mipso32.go
===================================================================
--- a/libgo/go/internal/syscall/unix/getrandom_linux_mipso32.go	(nonexistent)
+++ b/libgo/go/internal/syscall/unix/getrandom_linux_mipso32.go	(revision 240457)
@@ -0,0 +1,11 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build mipso32
+
+package unix
+
+// Linux getrandom system call number.
+// See GetRandom in getrandom_linux.go.
+const randomTrap uintptr = 4353
Index: b/libgo/go/internal/syscall/unix/getrandom_linux_sparc.go
===================================================================
--- a/libgo/go/internal/syscall/unix/getrandom_linux_sparc.go	(nonexistent)
+++ b/libgo/go/internal/syscall/unix/getrandom_linux_sparc.go	(revision 240457)
@@ -0,0 +1,11 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build sparc sparc64
+
+package unix
+
+// Linux getrandom system call number.
+// See GetRandom in getrandom_linux.go.
+const randomTrap uintptr = 347
Index: b/libgo/go/internal/syscall/unix/getrandom_linux_mips64x.go
===================================================================
--- a/libgo/go/internal/syscall/unix/getrandom_linux_mips64x.go	(revision 240456)
+++ b/libgo/go/internal/syscall/unix/getrandom_linux_mips64x.go	(revision 240457)
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build mips64 mips64le
+// +build mips64 mips64le mipsn64 mipso64
 
 package unix
 
Index: b/libgo/go/internal/syscall/unix/getrandom_linux_mipsn32.go
===================================================================
--- a/libgo/go/internal/syscall/unix/getrandom_linux_mipsn32.go	(nonexistent)
+++ b/libgo/go/internal/syscall/unix/getrandom_linux_mipsn32.go	(revision 240457)
@@ -0,0 +1,11 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build mipsn32
+
+package unix
+
+// Linux getrandom system call number.
+// See GetRandom in getrandom_linux.go.
+const randomTrap uintptr = 6317
